### ISSUE TITLE
[FEAT EXTRACTION]

### ISSUE THREAD

**Issue by Venshao** [Radiomics/pyradiomics#634]
**Describe the bug**
My dcm ImageArray shape is (512, 512, 1), as well as the mask, and it is nii.gz format. These mask files were converted from png to nii.gz, all metadata were copied from the original dcm. It seems to work well when I extract first-order features, but I encounter a fatal error when I attempt to extract GLCM features, detail logs see the log.

**PyRadiomics configuration**
```Python
    settings = dict()
    settings['binWidth'] = 25
    settings['force2D'] = True
    settings['correctMask'] = True
    settings['resampledPixelSpacing'] = [3, 3, 3]
    settings['interpolator'] = sitk.sitkBSpline```

**PyRadiomics log**

Traceback (most recent call last):
  File ".\features_extractor.py", line 87, in <module>
    x = extract(r'D:\B_all_data\fixed_set\train\1\1015\10011.dcm', r'D:\B_all_data\fixed_set\train\1\1015\10011_mask.nii.gz')
  File ".\features_extractor.py", line 80, in extract
    return extractor.execute(ir, mr, 255)
  File "H:\Programs\Anaconda3\lib\site-packages\radiomics\featureextractor.py", line 327, in execute
    featureVector.update(self.computeFeatures(inputImage, inputMask, imageTypeName, **inputKwargs))
  File "H:\Programs\Anaconda3\lib\site-packages\radiomics\featureextractor.py", line 514, in computeFeatures
    for (featureName, featureValue) in six.iteritems(featureClass.execute()):
  File "H:\Programs\Anaconda3\lib\site-packages\radiomics\base.py", line 185, in execute
    self._calculateSegment()
  File "H:\Programs\Anaconda3\lib\site-packages\radiomics\base.py", line 222, in _calculateSegment
    for success, featureName, featureValue in self._calculateFeatures():
  File "H:\Programs\Anaconda3\lib\site-packages\radiomics\base.py", line 229, in _calculateFeatures
    self._initCalculation(voxelCoordinates)
  File "H:\Programs\Anaconda3\lib\site-packages\radiomics\gldm.py", line 75, in _initCalculation
    self.P_gldm = self._calculateMatrix(voxelCoordinates)
  File "H:\Programs\Anaconda3\lib\site-packages\radiomics\gldm.py", line 103, in _calculateMatrix
    P_gldm = numpy.delete(P_gldm, emptyGrayLevels - 1, 1)
  File "<__array_function__ internals>", line 6, in delete
  File "H:\Programs\Anaconda3\lib\site-packages\numpy\lib\function_base.py", line 4406, in delete
    keep[obj,] = False
IndexError: arrays used as indices must be of integer (or boolean) type

**To Reproduce**
Steps to reproduce the behavior:
My dcm ImageArray shape is (512, 512, 1), as well as the mask, and it is nii.gz format. These mask files were converted from png to nii.gz, all metadata were copied from the original dcm. It seems to work well when I extract first-order features, but I encounter a fatal error when I attempt to extract GLCM features, detail logs see the log.

**Version (please complete the following information):**
 - OS: Win10
 - Python version: 3.6
 - PyRadiomics version: 3.0

**Comment by Venshao** [comment]
> Could it be due to the resampling? You use 3x3x3, but the image z axis only has 1 slice. As the extraction is 2d, I would try limiting resampling to the in plane resolution. 

I attempt to use 3x3x1, 1x3x3, 1x1x1, 3x1x3 respectively but got the same error.

**Comment by JoostJM** [comment]
Duplicate issue to #592, which is fixed by #594.
This is not yet part of the 3.0 version, but will be included in the next release.

**Comment by Venshao** [comment]
> Try setting `resampledPixelSpacing` to 0 for the dimension that corresponds to out-of-slice - this will do resampling in-plane only, such as 3x3x0.
Thank you for your reply, but I got the same error with 3x3x0

**Comment by Venshao** [comment]
> Duplicate issue to #592, which is fixed by #594.
> This is not yet part of the 3.0 version, but will be included in the next release.


I cloned the source code and modified glcm.py according to [MStarmans91](https://github.com/Radiomics/pyradiomics/issues/592#issuecomment-650270101 "MStarmans91") 's suggestion, add a condition to avoid this case.
![image](https://user-images.githubusercontent.com/10670390/92198121-26bb1c80-eea6-11ea-861f-ba60db9f3819.png)
Everything work perfectly.

**Comment by anjipas** [comment]
It seems this error comes from an update in numpy API. It seems in the new release (1.19) an error is thrown if emptyGrayLevels is an empty array.

**Comment by PaPablo** [comment]
Hi everyone, I solved this by downgrading numpy from `1.19` to `1.18` by running:

```bash
pip install numpy==1.18
```

It extracted everything fine.
